app/navigation/router.js

```diff
@@ -72,7 +75,7 @@ class Router extends React.Component {
                     renderLeftComponent={renderLeftComponent}
                     renderTitleComponent={renderTitleComponent}
                     renderRightComponent={renderRightComponent}
-                    style={navigationProps.headerStyle}
+                    style={[{backgroundColor: this.props.theme.sidebarHeaderBg}, navigationProps.headerStyle]}
```

Comment:
My thinking here was if you set the backgroundColor in the ```navigationProps.headerStyle``` then you would want it to override whatever is in the theme.